volumina/pixelpipeline/datasources/arraysource.py

```diff
@@ -77,14 +79,14 @@ def put(self, slicing, subarray, neutral=0):
         assert len(slicing) == len(
             self._array.shape
         ), "slicing into an array of shape=%r requested, but the slicing object is %r" % (slicing, self._array.shape)
-        self._array[slicing] = np.where(subarray != neutral, subarray, self._array[slicing])
+        self._array[slicing] = subarray
```

Comment:
Now we always write data that we previously read, so there is no "neutral" value.
Before
```python
src = array([0, 1, 1])
labels = array([0, 2, 0])
put_arr = array([0, 2, 0])
# result write only put_arr[1]
```
now
```python
src = array([0, 1, 1])
labels = array([0, 2, 0])
put_arr = array([0, 2, 1])
# result write whole put_arr
```

Reply:
right, that is not necessary anymore with your new implementation you do

```python
redo_data = numpy.where(self.__labels != 0, self.__labels, self.__old_data)
```
Before, you could draw, without requesting the data first and still only overpaint on stroke.

We should carefully chat, which code paths might also go to `put`...